ivy/func_wrapper.py

```diff
@@ -1300,7 +1336,8 @@ def _wrapped(func):
             if "frontends" in func.__module__:
                 # it's a frontend func, no casting modes for this
                 return func
-            return casting_modes_ops(func)
+
+            return casting_modes_ops(func, ret_dtype_target)
```

Comment:
let's remove the blank line above this and pass `ret_dtype_target` as a keyword argument